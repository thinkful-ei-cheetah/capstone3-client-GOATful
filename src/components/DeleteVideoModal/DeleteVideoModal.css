
 
 .delete-video-modal{
   max-width: 320px;
   min-width: 300px;
   border: 1px solid black;
   padding: 15px;
   border-radius: 5px;
   width: 60vw;
   height: 30vh;
   padding: 2em;
   position: relative;
   background:rgba(0, 0, 0, .7);
   color: white;
 }

 .cancel-btn{
  border: 1px solid white;
  background: none;
  box-shadow: 1px 1px white;
}

.cancel-btn:hover, .confirm-btn:hover{
 box-shadow: none;
}

.blank-space-delete{
  padding-bottom: 15px;
}

.delete-button-controls{
  padding-top: 20px;
 display: flex;
 justify-content: space-around
}
 
 @media screen and (max-width: 500px){
  .delete-video-modal{
    border: none;
    padding:20px;
    width: 300px;
    height: 40vh;     
   }
 }