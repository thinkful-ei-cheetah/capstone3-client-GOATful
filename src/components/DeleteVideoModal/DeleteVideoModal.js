import React, { Component } from 'react'
import Modal from 'react-modal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faWindowClose } from '@fortawesome/free-solid-svg-icons'
import './DeleteVideoModal.css'

const modalStyles = {
  content : {
    maxWidth              : '40%',
    width                 : '300px',
    top                   : '50%',
    left                  : '50%',
    right                 : 'auto',
    bottom                : 'auto',
    marginRight           : '-50%',
    transform             : 'translate(-50%, -50%)'
  }
};
Modal.setAppElement('#root')

export default class DeleteVideoModal extends Component {
  render() {
    return (
      <Modal
      isOpen={this.props.isOpen}
      onRequestClose={this.props.onRequestClose}
      style={modalStyles}
      contentLabel={'Delete Video'}
      closeTimeoutMS={200}
        >
          <h2 className="add-video-header">Are you sure?</h2>
          <p>Delete {this.props.fields.title}?</p>

          <p>{this.props.previewCount ? 
            this.props.previewCount === 1 ? `1 preview will also be deleted.` : `${this.props.previewCount} previews will also be deleted.` : '' }</p>
         
         <span className='close-modal-btn' onClick={this.props.onRequestClose}>
            <FontAwesomeIcon 
              icon={faWindowClose}
            />
          </span>
          <div className= "button-controls">
            <button className="cancel" onClick={this.props.onRequestClose}>Cancel</button>
            <button className="confirm" onClick={this.props.handleConfirm}>Delete</button>
          </div>
      </Modal>
    )
  }
}
