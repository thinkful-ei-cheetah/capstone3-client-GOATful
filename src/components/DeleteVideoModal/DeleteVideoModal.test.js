import React from 'react';
import ReactDOM from 'react-dom';
import DeleteVideoModal from './DeleteVideoModal';

describe('DeleteVideoModal Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< DeleteVideoModal />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})
