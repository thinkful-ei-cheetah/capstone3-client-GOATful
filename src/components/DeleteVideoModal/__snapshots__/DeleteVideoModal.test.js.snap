// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteVideoModal Component renders the UI as expected 1`] = `null`;
